src/diffusers/models/unets/unet_i2vgen_xl.py

```diff
@@ -147,10 +148,16 @@ def __init__(
         layers_per_block: int = 2,
         norm_num_groups: Optional[int] = 32,
         cross_attention_dim: int = 1024,
+        attention_head_dim: Union[int, Tuple[int]] = None,
```

Comment:
I think it might be better to set `attention_head_dim` to 64 and update the I2VGenXL config? And then in the code we can just ignore the passed in `num_attention_heads` from the config.

Reply:
Elaborate why?

Reply:
I think it is ok to set `attention_head_dim` to be `64` default but we can't just ignore `num_attention_heads`. would still need to deprecate it properly so our user knows that they should stop using it

Reply:
This keeps getting more confusing :D 

From what @DN6 suggested internally:

> Ideally, I would like to keep only one of these params when we get 1.0 (num_attention_heads IMO). I think for the 2D UNets it's possible to find a way to only use num_attention_heads for all future 2D UNets, while for the 3D UNet, we should first clean up the naming and then figure out the best way to only use num_attention_heads correctly. 

How can we then get rid of `num_attention_heads`?

Reply:
Here's what I'm proposing:

1. Add `attention_head_dim` to the I2VGenXL UNet, set it to `64` while also updating the config on the hub so that it has an `attention_head_dim` parameter. This is because, currently, in the I2VGenXL config on the hub, we have a parameter called `num_attention_heads` which actually means `attention_head_dim`.  We want to avoid using names that mean something else in the source code and avoid lines like this where possible
```
attention_head_dim = num_attention_heads or attention_head_dim
```
2. After updating the config and I2VGenXL UNet to accept `attention_head_dim`, internally in the code we use `attention_head_dim`  to directly compute the proper value of `num_attention_heads` and pass it to the get_down_blocks and get_up_block functions to the 3D blocks. And update the blocks themselves to correctly use this value. We should should do the same for the 3D UNet. 

We want to remove stuff like 
```attention_head_dim=num_attention_heads```
or 
```num_attention_heads=attention_head_dim```

Currently the way it is done in the 3D blocks is very confusing. Here's what we are currently doing:

We set `num_attention_heads` to the value of `attention_head_dim` 
https://github.com/huggingface/diffusers/blob/76696dca558267999abf3e7c29e1a256cbcb407a/src/diffusers/models/unets/unet_3d_condition.py#L141

and then inside the blocks we use it in a weird way e.g  (https://github.com/huggingface/diffusers/blob/76696dca558267999abf3e7c29e1a256cbcb407a/src/diffusers/models/unets/unet_3d_blocks.py#L358-L360)

The first argument to Transformer2D is supposed to be `num_attention_heads` , but we're dividing the `out_channels`  by the `num_attention_heads` because we've set it to mean `attention_head_dim`, and then we pass in `num_attention_heads` to the `attention_head_dim` argument. This is very confusing. These UNet 3D Blocks needs to change so that they work the same way as the 2D blocks: 

https://github.com/huggingface/diffusers/blob/76696dca558267999abf3e7c29e1a256cbcb407a/src/diffusers/models/unets/unet_2d_blocks.py#L721-L722) 

where the parameter names and values are aligned. 

None of the UNet blocks accept an `attention_head_dim` argument. So I think we should aim to make `num_attention_heads` the single parameter that is accepted by the UNet models, and compute the `attention_head_dim` where necessary using this parameter.  

3. Adding `attention_head_dim` to all the blocks would add to further confusion. And we can't just remove it from the 3D UNet completely because it would be backwards breaking for text2vid synth.  

4. Regarding I2VGenXL, by adding `attention_head_dim` and just ignoring the incorrectly defined `num_attention_heads` parameter for future versions, the thing we're mostly fixing is clarity around what the parameter does in the source code by avoiding the use of an incorrectly defined parameter. 

Fixing the issue for this version might not even be possible because if we try to change the `num_attention_heads` value in the hub config, it will break for everyone using version `0.26`.

Reply:
Okay I think I understand. What you're suggesting would need changes to:

* https://github.com/huggingface/diffusers/blob/main/src/diffusers/models/unets/unet_3d_blocks.py (changes to `get_down_block()` and `get_up_block()`) so that pt. 2 is reflected. 
* We don't touch https://github.com/huggingface/diffusers/blob/main/src/diffusers/models/unets/unet_3d_condition.py to respect pt. 3.
* https://github.com/huggingface/diffusers/blob/main/src/diffusers/models/unets/unet_i2vgen_xl.py to reflect pt. 1, 2, and 4.

Am I thinking correctly?